### PR TITLE
Enable testing of ImageNet and MNIST models from classify.py

### PR DESCRIPTION
**Pull request by dribnet**
This simply re-introduces changes originally added in #735 by @petewarden to the current version of caffe.

This enables testing the ImageNet Caffe model from classify.py as described in [the current wiki pages](https://github.com/BVLC/caffe/wiki/Ubuntu-14.04-VirtualBox-VM):

``` bash
$ python python/classify.py --print_results examples/images/cat.jpg foo

Loading file: examples/images/cat.jpg
Classifying 1 inputs.
Done in 1.43 s.
[('tabby', '0.27353'), ('tiger cat', '0.23681'), ('Egyptian cat', '0.15175'), ('lynx', '0.12904'), ('kit fox', '0.04705')]
Saving results into foo
```

and also adds MNIST support:

```
$ ./train_lenet.sh
$ python python/classify.py --print_results \
  --model_def examples/mnist/lenet.prototxt \
  --pretrained_model examples/mnist/lenet_iter_10000.caffemodel \
  --force_grayscale --center_only \
  --labels_file data/mnist/mnist_words.txt \
  --images_dim 28,28 data/mnist/sample_digit.png resultsfile

Loading image file: data/mnist/sample_digit.png
Classifying 1 inputs.
Done in 0.05 s.
[('Four', '1.00000'), ('Zero', '0.00000'), ('One', '0.00000'), ('Two', '0.00000'), ('Three', '0.00000')]
Saving results into resultsfile
```

### CODE REVIEW

**Comment by kkarnatak** [comment]
@dribnet I am getting an error `NameError: global name 'scores' is not defined` while executing classify.py using --print_results argument. I have followed all the steps as suggested in your first post on this thread, and have also googled elsewhere, but no luck. Did you face this kinda issue as well?

**Comment by kkarnatak** [comment]
@dribnet I got it. I used the files associated with this thread and replaced "scores" with "predictions".

For reference:

 indices = (-predictions[0]).argsort()[:5]
 predicted_labels = labels[indices]

Thanks.

**Comment by shelhamer** [comment]
Closing as `caffe.Classifier` is soon to be deprecated. Thanks for polishing the script as it was all the same.

File: python/classify.py [hunk 1]
```diff
@@ -129,6 +148,27 @@ def main(argv):
...
+    if args.print_results:
+        with open(args.labels_file) as f:
+          labels_df = pd.DataFrame([
+               {
+                   'synset_id': l.strip().split(' ')[0],
+                   'name': ' '.join(l.strip().split(' ')[1:]).split(',')[0]
+               }
+               for l in f.readlines()
+            ])
+        labels = labels_df.sort('synset_id')['name'].values
```
Comment (rajasekharponakala) [inline python/classify.py]: deprecated use of `sort`, now changed to `sort_values`.
~
http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.sort_values.html

**Pull request closed without merging**